### PR TITLE
DAOS-5663 control: Add rich errors for MS raft

### PR DESCRIPTION
**Pull request by mjmac**
Encode MS leadership/replica info in new error types that
survive the trip over gRPC.

Also moves the component name variables into the build
package, which is a better location for them.

### CODE REVIEW

**Review by tanabarr — DISMISSED** [review]

**Review by jbrosenz — APPROVED** [review]
LGTM

File: src/control/common/proto/error_test.go [hunk 1]
```diff
@@ -148,27 +123,29 @@ func TestProto_AnnotateError(t *testing.T) {
...
 			aErr := proto.AnnotateError(tc.err)
 
-			var gotErr error
+			gotErr := proto.UnwrapError(status.Convert(aErr))
+			common.CmpErr(t, tc.expErr, gotErr)
+			if tc.expErr == nil {
+				return
+			}
+
 			switch tc.err.(type) {
```
Comment (kjacque) [inline src/control/common/proto/error_test.go:132]: Is this switch block still needed? Looks like the expErr/gotErr comparison above should cover this, should it not?

File: src/control/system/errors.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/pkg/errors"
+)
+
+// ErrNotReplica indicates that a request was made to a control plane
+// instance that is not a designated Management Service replica.
+type ErrNotReplica struct {
+	Replicas []string
+}
+
+func (err *ErrNotReplica) Error() string {
```
Comment (kjacque) [inline src/control/system/errors.go:40]: I know it's self-explanatory, but does the linter expect a comment for it?

**Review by kjacque — DISMISSED** [review]
Only a couple minor questions/comments.

File: src/control/system/errors.go (hunk 2 above)
Comment (mjmac) [reply src/control/system/errors.go:40]: Nope, implementations of error and fmt.Stringer don't require comments.

File: src/control/common/proto/error_test.go (hunk 1 above)
Comment (mjmac) [reply src/control/common/proto/error_test.go:132]: Hmm, good question. I think I cargo-culted my own code without really thinking about it. There was some goofy business in there before with the default case... Can probably clean this up to make it easier to maintain.

File: src/control/system/errors.go (hunk 2 above)
Comment (kjacque) [reply src/control/system/errors.go:40]: Interesting to know! Thanks.

**Comment by mjmac** [comment]
Known failures:
  - [DAOS-5591](https://jira.hpdd.intel.com/browse/DAOS-5591)
  - [DAOS-4943](https://jira.hpdd.intel.com/browse/DAOS-4943)

**Pull request merged**